ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -1008,16 +1008,19 @@ public boolean restoreState(InputObjectState os)
 	                        }
 	                    }
 
-	                    // Give it a go ourselves
-	                    if (!deserialized) {
-	                        _theXAResource = (XAResource) o.readObject();
-	                    }
-
-	                    o.close();
-
-	                    if (jtaxLogger.logger.isTraceEnabled()) {
-	                        jtaxLogger.logger.trace("XAResourceRecord.restore_state - XAResource de-serialized");
-	                    }
+                        try {
+                            // Give it a go ourselves
+                            if (!deserialized) {
+                                _theXAResource = (XAResource) o.readObject();
+                            }
+                            if (jtaxLogger.logger.isTraceEnabled()) {
+                                jtaxLogger.logger.trace("XAResourceRecord.restore_state - XAResource de-serialized");
+                            }
+                        } catch (ClassNotFoundException cnfe) {
+                            jtaxLogger.i18NLogger.warn_could_not_load_class_will_wait_for_bottom_up(cnfe);
```

Comment:
Just a remark: am I correct in assuming the other exception types thrown by o.readObject are unrecoverable and therefore we must not fallback to using bottom up recover?  Do the other exception types indicate either mis-configuration or object store corruption?